internal/transport/http2_client.go

```diff
@@ -1327,30 +1359,25 @@ func (t *http2Client) operateHeaders(frame *http2.MetaHeadersFrame) {
 		}
 	}
 
-	if !isGRPC {
-		var (
-			code           = codes.Internal // when header does not include HTTP status, return INTERNAL
-			httpStatusCode int
-		)
-
-		if httpStatus != "" {
-			c, err := strconv.ParseInt(httpStatus, 10, 32)
-			if err != nil {
-				se := status.New(codes.Internal, fmt.Sprintf("transport: malformed http-status: %v", err))
-				t.closeStream(s, se.Err(), true, http2.ErrCodeProtocol, se, nil, endStream)
-				return
-			}
-			httpStatusCode = int(c)
+	if !isGRPC || httpStatusErr != "" {
+		var code = codes.Internal // when header does not include HTTP status, return INTERNAL
 
+		if httpStatusCode != nil {
 			var ok bool
-			code, ok = HTTPStatusConvTab[httpStatusCode]
+			code, ok = HTTPStatusConvTab[*httpStatusCode]
```

Comment:
Interesting.  I just was pointed to this:

https://github.com/grpc/grpc/blob/master/doc/http-grpc-status-mapping.md

> This table is to be used only for clients that received a response that did not include grpc-status. If grpc-status was provided, it must be used


So, I think this part should only be happening if there is no grpc-status (rawStatusCode), but using rawStatusCode otherwise.  Do you want to take a shot at this?

Reply:
I've put up a PR for if [here](https://github.com/grpc/grpc-go/pull/4606)